src/Algebra/Structures/Biased.agda

```diff
@@ -0,0 +1,196 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Ways to give instances of certain structures where some fields can
+-- be given in terms of others
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Relation.Binary using (Rel; Setoid; IsEquivalence)
+
+module Algebra.Structures.Biased
+  {a ℓ} {A : Set a}  -- The underlying set
+  (_≈_ : Rel A ℓ)    -- The underlying equality relation
+  where
+
+open import Algebra.Core
+open import Algebra.Definitions _≈_
+open import Algebra.Structures _≈_
+import Algebra.FunctionProperties.Consequences as Consequences
+open import Data.Product using (_,_; proj₁; proj₂)
+open import Level using (_⊔_)
+
+------------------------------------------------------------------------
+-- IsCommutativeMonoid
+
+record IsCommutativeMonoidˡ (∙ : Op₂ A) (ε : A) : Set (a ⊔ ℓ) where
+  field
+    isSemigroup : IsSemigroup ∙
+    identityˡ   : LeftIdentity ε ∙
+    comm        : Commutative ∙
+
+  open IsSemigroup isSemigroup
+
+  private
+
+    identityʳ : RightIdentity ε ∙
+    identityʳ = Consequences.comm+idˡ⇒idʳ setoid comm identityˡ
+
+    identity : Identity ε ∙
+    identity = (identityˡ , identityʳ)
+
+  isCommutativeMonoid : IsCommutativeMonoid ∙ ε
+  isCommutativeMonoid = record
+    { isMonoid = record
+      { isSemigroup = isSemigroup
+      ; identity    = identity
+      }
+    ; comm = comm
+    }
```

Comment:
Could we publicly re-export the latter as e.g. `isCommutativeMonoidˡ`? That would mean users having to adjust existing code wouldn't have to prefix them with `IsCommutativeMonoidˡ`...

Reply:
I've come round to this, so it's in now.